src/native/minipal/gethomedir.h

```diff
@@ -0,0 +1,40 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#ifndef HAVE_MINIPAL_GETHOMEDIR_H
+#define HAVE_MINIPAL_GETHOMEDIR_H
+
+#include <errno.h>
+#include <pwd.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// Returns the full path to the home directory.
+// The caller is responsible for releasing the buffer. Returns null on error.
+static inline char* minipal_gethomedir(void)
+{
+    char* homedir = getenv("HOME");
+    if (!homedir)
+    {
+        uid_t uid = getuid();
+        struct passwd* pwuid = NULL;
+        while ((pwuid = getpwuid(uid)) == NULL && errno == EINTR);
```

Comment:
This isn't thread-safe, is it?  The man page states:
```
The return value may point to a static area, and may be overwritten by subsequent calls to getpwent(3), getpwnam(), or getpwuid().
```

Reply:
Correct. I should have used `getpwuid_r`.